### DISCUSSION TITLE
Extracting features from a git repo with Kaiaulu

### DISCUSSION THREAD

**Discussion by BenjyNStrauss** [sailuh/kaiaulu#311] [Q&A] [answered]
Which of the following data points can be extracted from a git repo with Kaiaulu?  And for those that can be extracted, could you provide the instructions and/or a link to them?

• total # of outstanding bugs
• total # of outstanding non-bugs (typically feature requests)

The following are for a given period of time:
• # of new bug issues
• # of new non-bug issues
• average bug-resolution time
• average non-bug resolution time
• # of active contributors
• # of new contributors
• # of bug-fixing commits
• # of non-bug-fixing commits
• # of LOC committed for bug resolution
• # of LOC committed for non-bug resolution
• # of emails on the project mailing list

**Comment by carlosparadis** [comment, marked as answer]
Hi Beni, @rnkazman 

The way to go about Kaiaulu is asking yourself first "where is the data coming from?" 

For bug data, that means you need to collect issue tracker data first. Then the question is, from which issue tracker? Kaiaulu can get you data from JIRA, GItHub, and Bugzilla. 

Once you decide on that, you can go to the respective menu on the "Reference" page for any of these:

http://itm0.shidler.hawaii.edu/kaiaulu/reference/index.html#-jira-

and subsequently, see the Notebook to obtain the data. From the table obtained, you can then calculate any metrics you wish.

Since you are interested in calculating a **Metric** out of the data above, then you can click on the "Metric" menu on the right: 

http://itm0.shidler.hawaii.edu/kaiaulu/reference/index.html#-metrics-

You will find there is a Bug Count Notebook there.

----

The same process can be used for your other metrics. For example, if you want to calculate contributor metrics, you can again ask yourself "What is the data that I need to obtain this metric?"

That would be "Git". Again, you can go to the docs page above and click "Git" and see the associated Git Log table. Same with Mail, etc. Same with "Mail".  

----

As for how to link them: Depends on what you want to link them on, and what granularity. If you can give me something more specific, I can give you pointers. 

The bottomline is: Kaiaulu will give you tables, and for most of them you will be making inner joins out of them to link. If you plan to connect people, see the "Identity" section and the associated Notebook. 

If you are looking for a one button press solution to create the metrics above, we do not have that. But it should be relatively simple (group by, subset, inner joins) to get to them from the tables Kaiaulu gives you.
